object_storage_api/repositories/image.py

```diff
@@ -49,13 +50,21 @@ def get(self, image_id: str, session: ClientSession = None) -> Optional[ImageOut
         :param image_id: ID of the image to retrieve.
         :param session: PyMongo ClientSession to use for database operations.
         :return: Retrieved image or `None` if not found.
+        :raises MissingRecordError: If the supplied image_id is invalid or non-existent.
         """
-        image_id = CustomObjectId(image_id)
         logger.info("Retrieving image with ID: %s from the database", image_id)
-        image = self._images_collection.find_one({"_id": image_id}, session=session)
+        try:
+            image_id = CustomObjectId(image_id)
+            image = self._images_collection.find_one({"_id": image_id}, session=session)
+        except InvalidObjectIdError as e:
+            exc = MissingRecordError(f"The image_id {image_id} is not valid.")
+            exc.response_detail = f"Invalid image_id format: {image_id}"
+            raise exc from e
         if image:
             return ImageOut(**image)
-        return None
+        exc = MissingRecordError(f"Image with image_id {image_id} was not found.")
+        exc.response_detail = f"Image with image_id {image_id} was not found."
+        raise exc
```

Comment:
You can do what we do in IMS and have
```message = f"Image with image_id {image_id} was not found."```
above and then use it in both exceptions. Same for the response detail.